src/components/legend/defaults.js

```diff
@@ -0,0 +1,70 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var Plots = require('../../plots/plots');
+
+var attributes = require('./attributes');
+var helpers = require('./helpers');
+
+
+module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
+    var containerIn = layoutIn.legend || {},
+        containerOut = layoutOut.legend = {};
+
+    var visibleTraces = 0,
+        defaultOrder = 'normal',
+        trace;
+
+    for(var i = 0; i < fullData.length; i++) {
+        trace = fullData[i];
```

Comment:
I prefer having: `var trace` inside the block.

The diff :arrow_double_up: is just showing old code being moved around. I'll change this right up.